src/media/js/apps_buttons.js

```diff
@@ -47,29 +48,23 @@ define('apps_buttons',
 
         // If it's a paid app, ask the user to sign in first.
         if (product.receipt_required && !user.logged_in()) {
-            var payWindow;
+            console.log('Install suspended; user needs to log in');
 
-            if (!capabilities.navPay) {
-                console.log('Creating payment window ahead of time via user click');
-                payWindow = payments.utils.openWindow({url: ''});
-            }
+            // Create a blank window here so we can pass it to the login func;
+            loginPopup = (!capabilities.navPay) ? utils.openWindow() : undefined;
 
-            console.log('Install suspended; user needs to log in');
-            return login.login().done(function() {
-                if (payWindow) {
-                    payWindow.focus();
-                }
+            return login.login({retainPopup: true, popupWindow: loginPopup}).done(function() {
                 // Once login completes, just call this function again with
                 // the same parameters, but re-fetch the button (since the
                 // button instance is not the same).
                 var new_button = get_button(product.manifest_url);
-                install(product, new_button);
+                install(product, new_button, loginPopup);
             }).fail(function(){
-                if (payWindow) {
-                    payWindow.close();
-                }
                 console.log('Install cancelled; login aborted');
                 notification.notification({message: gettext('Payment cancelled.')});
+                if (loginPopup) {
+                    loginPopup.close();
+                }
```

Comment:
I feel like this should abort the rest of the install() function otherwise ``payments.purchase()`` will still be called later on. Install is a really big function :(